libraries/rush-lib/src/logic/operations/Operation.ts

```diff
@@ -148,4 +154,83 @@ export class Operation {
     (this.dependencies as Set<Operation>).delete(dependency);
     (dependency.consumers as Set<Operation>).delete(this);
   }
+
+  public get stateHash(): string {
+    if (!this._stateHash) {
+      throw new Error(
+        'Operation state hash is not calculated yet, you must call `calculateStateHash` first.'
+      );
+    }
+    return this._stateHash;
+  }
```

Comment:
This information is pertinent to a _single_ execution of an `Operation`, so belongs on `OperationExecutionRecord` (that's the job of said class). I'm trying to get us to a world where in watch mode (or other service-like mode), the graph of `Operation`s is created once and they are executed repeatedly (with fresh `OperationExecutionRecord` instances to reflect the status of the latest execute).